### PR TITLE
refactor(journal): account typeahead opens empty on focus

### LINKED ISSUE #1545: (feature request)  Autodropdown and autofocus account input on edit Posting Journal
As mentioned in #1541:
> Pour editer un compte, la cas doit etre selectionnez apres double clique pour que la recheche commence avec des chiffres entrees du claivier; preferablement avec drop down ouvert sur l'ancien compte.

Using the `ui-bootstrap` [typeahead](https://angular-ui.github.io/bootstrap/#!#typeahead)'s `min-length` property, it should be possible to pop the dropdown menu up before the user has typed anything.

### PR DESCRIPTION
**Pull request by jniles**
A request from the accountant using the system is for the account typeahead to open automatically on
focus, for the input to autofocus, and for the input to be cleared.  This commit implements those as
best as possible with fickle HTML and typeahead API.

Closes #1545.

---

Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through ESLint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

**Comment by jniles** [comment]
Oops, I apparently didn't fork properly.  I'll rebase this.

File: client/src/modules/journal/modals/trialBalanceDetail.footer.html [hunk 1]
```diff
@@ -1,14 +1,9 @@
-<button
-  id="cancelID"
-  type="button"
-  class="btn btn-default"
-  ng-click="TrialBalanceFooterDetailCtrl.cancel()">
+<button type="button" class="btn btn-default" data-method="cancel" ui-sref="journal">
```
Comment (sfount) [inline client/src/modules/journal/modals/trialBalanceDetail.footer.html:1]: :+1:

File: client/src/modules/journal/ui-grid-edit-account.directive.js [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
     restrict : 'A',
     template : function () {
       var templ =
         '<div style="display: table; margin-right: auto; margin-left: auto; width:95%">' +
-        '  <input type="text"' +
+        '  <input type="text" autofocus ' +
         '    style="width:100%" ' +
         '    ng-model="accountInputValue" ' +
         '    uib-typeahead="account.id as account.hrlabel for account in accounts | filter:{\'hrlabel\':$viewValue} | limitTo:8" ' +
+        '    typeahead-min-length="0" ' +
```
Comment (sfount) [inline client/src/modules/journal/ui-grid-edit-account.directive.js:16]: This is a very interesting solution!

**Review by sfount — APPROVED** [review]
The code here LGTM, I haven't fetched and tested the functionality but the entire Journal will be reviewed soon; that should cover this.

**Pull request merged**